### DISCUSSION TITLE
Problems with battery health dashboard

### DISCUSSION THREAD

**Discussion by inigoml** [jheredianet/Teslamate-CustomGrafanaDashboards#50] [Q&A] [answered]
Hi everybody.

I installed Custom Dashboards barely a month ago. All dashboards work properly (good job, by the way) but the Battery Health.
This is an screenshow I get each time I access this dashboard.

<img width="1122" alt="image" src="https://github.com/jheredianet/Teslamate-CustomGrafanaDashboards/assets/7486708/48d1d4d7-90ec-47bb-a024-379a0d37fc0e">
 
And the error is always this:
`
db query error: pq: invalid input syntax for type json`

for queries like this:
```
SELECT  
  CASE WHEN 80 > 0 THEN 80 ELSE (''::json -> 'MaxCapacity')::text::float END as "Usable (new)", 
  (''::json -> 'CurrentCapacity')::text::float as "Usable (now)",
  (''::json -> 'CurrentCapacity')::text::float - CASE WHEN 80 > 0 THEN 80 ELSE (''::json -> 'MaxCapacity')::text::float END as "Difference"

```
What am I missing?

I'm not raising a bug since probably its something related with my installation.

**Comment by inigoml** [comment]
By the way, with my previous car there is no problem. I sold it one year ago but logged data remains.

<img width="1152" alt="image" src="https://github.com/jheredianet/Teslamate-CustomGrafanaDashboards/assets/7486708/b54dfb9d-874f-4185-a59d-af5f30f715a4">

And take a look at the difference between both queries (first and second car)

```
SELECT  
  CASE WHEN 55 > 0 THEN 55 ELSE ('{"LastChargekWhAdded": 2.96, "LastMileageAdded" : 23.29, "MaxRange": 416.2712805403563600, "CurrentRange": 405.6043773511163200, "MaxCapacity": 53.236491803278696, "CurrentCapacity": 52.3732534493076255, "RatedEfficiency": 12.740000000000002}'::json -> 'MaxCapacity')::text::float END as "Usable (new)", 
  ('{"LastChargekWhAdded": 2.96, "LastMileageAdded" : 23.29, "MaxRange": 416.2712805403563600, "CurrentRange": 405.6043773511163200, "MaxCapacity": 53.236491803278696, "CurrentCapacity": 52.3732534493076255, "RatedEfficiency": 12.740000000000002}'::json -> 'CurrentCapacity')::text::float as "Usable (now)",
  ('{"LastChargekWhAdded": 2.96, "LastMileageAdded" : 23.29, "MaxRange": 416.2712805403563600, "CurrentRange": 405.6043773511163200, "MaxCapacity": 53.236491803278696, "CurrentCapacity": 52.3732534493076255, "RatedEfficiency": 12.740000000000002}'::json -> 'CurrentCapacity')::text::float - CASE WHEN 55 > 0 THEN 55 ELSE ('{"LastChargekWhAdded": 2.96, "LastMileageAdded" : 23.29, "MaxRange": 416.2712805403563600, "CurrentRange": 405.6043773511163200, "MaxCapacity": 53.236491803278696, "CurrentCapacity": 52.3732534493076255, "RatedEfficiency": 12.740000000000002}'::json -> 'MaxCapacity')::text::float END as "Difference"


SELECT  
  CASE WHEN 55 > 0 THEN 55 ELSE (''::json -> 'MaxCapacity')::text::float END as "Usable (new)", 
  (''::json -> 'CurrentCapacity')::text::float as "Usable (now)",
  (''::json -> 'CurrentCapacity')::text::float - CASE WHEN 55 > 0 THEN 55 ELSE (''::json -> 'MaxCapacity')::text::float END as "Difference"

```

**Reply by inigoml** [reply]
Could it be related with deployment script that only takes into consideration first car and not second card??

**Reply by jheredianet** [reply, marked as answer]
First of all, Are you using the last version of the dashboard?
No matter about having two or more cars, becase variables are set on dashboard load or car changed. 
Maybe you should press F5 to refresh the browser and try again. 
If no data is shown, it's because you need at least a long charge session and drives to get the estimations. 
About this error:  "db query error: pq: invalid input syntax for type json" it's because of that, you need more data logged, to estimate.
